DoD and Numbering format

smendis-scottlogic opened the pull request.

damienjburks commented on docs/governance/community-guidelines/numbering-format.md:
```diff
@@ -0,0 +1,13 @@
+- **TX** - Taxonomy. Fixed
+- **CAT** - Refer to Category. Valid values NW (Network Services), COMP (Computing Services), DB (Database Services), STR (Storage Services)
+- **XX** - Service Number within the Category. Numbering will specify the order we start working on.
+- **YYY** - Feature Number within the Service. Numbering will not decide a prority order.
```
```suggestion
- **YYY** - Feature Number within the Service. Numbering will not decide a priority order.
```

damienjburks commented on docs/governance/community-policies/definition-of-done.md:
```diff
@@ -0,0 +1,14 @@
+- All identified features should be numbered according to the _common-cloud-control_ guidelines.
+- A description should be provided for all identified features.
+- An asset describing the ways to test the presence of each control should be documented in feature file with the name `<service-name>.feature`.
+- All higher-level groupings in the hierarchy should be completed with `taxonmy.md` and `<service-name>.feature` files of its own containing the common features.
```
```suggestion
- All higher-level groupings in the hierarchy should be completed with `taxonomy.md` and `<service-name>.feature` files containing the common features.
```

damienjburks reviewed (commented):
Left two suggestions for this. I think we should move the numbering-format.md file into the community policies folder instead. Thoughts on this?

eddie-knight commented on docs/governance/community-guidelines/numbering-format.md:
```diff
@@ -0,0 +1,13 @@
+
+- **CCC** - Common Coloud Control. Fixed
+- **TX** - Taxonomy. Fixed
+- **CAT** - Refer to Category. Valid values NW (Network Services), COMP (Computing Services), DB (Database Services), STR (Storage Services)
```
```suggestion
- **CAT** - Refer to Category. Valid values: NW (Network Services), COMP (Computing Services), DB (Database Services), STR (Storage Services)
```

Is this a comprehensive list?

smendis-scottlogic replied:
No. These are the only categories we have started work on at the moment

eddie-knight commented on docs/governance/community-guidelines/numbering-format.md:
```diff
@@ -0,0 +1,13 @@
+- **CCC** - Common Coloud Control. Fixed
+- **TX** - Taxonomy. Fixed
+- **CAT** - Refer to Category. Valid values NW (Network Services), COMP (Computing Services), DB (Database Services), STR (Storage Services)
+- **XX** - Service Number within the Category. Numbering will specify the order we start working on.
```
```suggestion
- **XX** - Service Number within the Category. Numbering will dictate the sequence of work.
```

eddie-knight commented on docs/governance/community-guidelines/numbering-format.md:
```diff
@@ -0,0 +1,13 @@
+- **TX** - Taxonomy. Fixed
+- **CAT** - Refer to Category. Valid values NW (Network Services), COMP (Computing Services), DB (Database Services), STR (Storage Services)
+- **XX** - Service Number within the Category. Numbering will specify the order we start working on.
+- **YYY** - Feature Number within the Service. Numbering will not decide a priority order.
```
```suggestion
- **YYY** - Feature Number within the Service. Numbering will not decide a priority.
```

eddie-knight commented on docs/governance/community-policies/definition-of-done.md:
```diff
@@ -0,0 +1,14 @@
+# Definition of Done (DoD) for a Service
```
Is this intended to include the end-to-end delivery of a service, both taxonomy definition and control definitions?

smendis-scottlogic replied:
No this was written with only the taxonomy in mind. Happy if someone can extend this to cover control definition

eddie-knight commented on docs/governance/community-policies/definition-of-done.md:
```diff
@@ -0,0 +1,14 @@
+
+Service in this context is anything that can not be further categorized into subcategories. For example _Relational Databases_ is a service where _Databases_ is considered as a higher level grouping of common features among all different types of databases.
+
+Stated below is the set of criteria that a service must meet for the team to consider it complete and ready for its consumers.
```
```suggestion
Stated below is the set of criteria that a service must meet to be ready for delivery.
```

eddie-knight commented on docs/governance/community-policies/definition-of-done.md:
```diff
@@ -0,0 +1,14 @@
+
+Stated below is the set of criteria that a service must meet for the team to consider it complete and ready for its consumers.
+
+- A comprehensive set of minimum viable features should be identified for the service.
```
```suggestion
- A comprehensive set of minimum viable features has been identified for the service.
```

changing phrasing to past tense (as in, this must be true at the time it is considered done)

eddie-knight commented on docs/governance/community-policies/definition-of-done.md:
```diff
@@ -0,0 +1,14 @@
+Stated below is the set of criteria that a service must meet for the team to consider it complete and ready for its consumers.
+
+- A comprehensive set of minimum viable features should be identified for the service.
+- All possible higher-level groupings of features should be identified for the service.
```
```suggestion
- All possible higher-level groupings of features have been defined for the service.
```

eddie-knight commented on docs/governance/community-policies/definition-of-done.md:
```diff
@@ -0,0 +1,14 @@
+
+- A comprehensive set of minimum viable features should be identified for the service.
+- All possible higher-level groupings of features should be identified for the service.
+- A folder should be created with the name of the service, under **services** folder in the repository which clearly depicts its place in the taxonomy. E.g. `services > databases > relational` where _relational_ is the service name and _databases_ is the higher-level grouping
```
```suggestion
- A folder has been created under **services** folder, with the name of the service, according to its place in the taxonomy. (e.g. `services/databases/relational/` where _relational_ is the service name and _databases_ is the higher-level grouping)
```

eddie-knight commented on docs/governance/community-policies/definition-of-done.md:
```diff
@@ -0,0 +1,14 @@
+- A comprehensive set of minimum viable features should be identified for the service.
+- All possible higher-level groupings of features should be identified for the service.
+- A folder should be created with the name of the service, under **services** folder in the repository which clearly depicts its place in the taxonomy. E.g. `services > databases > relational` where _relational_ is the service name and _databases_ is the higher-level grouping
+- All identified features should be documented in a `taxonomy.md` file under the folder created for the service.
```
```suggestion
- All identified have been documented in a `taxonomy.md` file within the directory created for the service.
```

eddie-knight commented on docs/governance/community-policies/definition-of-done.md:
```diff
@@ -0,0 +1,14 @@
+- All possible higher-level groupings of features should be identified for the service.
+- A folder should be created with the name of the service, under **services** folder in the repository which clearly depicts its place in the taxonomy. E.g. `services > databases > relational` where _relational_ is the service name and _databases_ is the higher-level grouping
+- All identified features should be documented in a `taxonomy.md` file under the folder created for the service.
+- All identified features should be numbered according to the _common-cloud-control_ guidelines.
```
```suggestion
- All common features have been numbered according to the (numbering format)[../numbering-format.md].
```

eddie-knight commented on docs/governance/community-policies/definition-of-done.md:
```diff
@@ -0,0 +1,14 @@
+- A folder should be created with the name of the service, under **services** folder in the repository which clearly depicts its place in the taxonomy. E.g. `services > databases > relational` where _relational_ is the service name and _databases_ is the higher-level grouping
+- All identified features should be documented in a `taxonomy.md` file under the folder created for the service.
+- All identified features should be numbered according to the _common-cloud-control_ guidelines.
+- A description should be provided for all identified features.
```
```suggestion
- A description has been provided for all common features.
```

eddie-knight commented on docs/governance/community-policies/definition-of-done.md:
```diff
@@ -0,0 +1,14 @@
+- All identified features should be documented in a `taxonomy.md` file under the folder created for the service.
+- All identified features should be numbered according to the _common-cloud-control_ guidelines.
+- A description should be provided for all identified features.
+- An asset describing the ways to test the presence of each control should be documented in feature file with the name `<service-name>.feature`.
```
```suggestion
- An gherkin feature file describing the ways to test the presence of each control has been cretated in the service directory with the name `<service-name>.feature`.
```

eddie-knight replied:
It looks as if we have diverged on the naming convention between the `.md` and `.feature` files. 

What do you think of using a unique name for each taxonomy file, similar to the gherkin? Alternatively it might be the other way around, with generic names for the feature files within a service directory.

eddie-knight commented on docs/governance/community-policies/definition-of-done.md:
```diff
@@ -0,0 +1,14 @@
+- All identified features should be numbered according to the _common-cloud-control_ guidelines.
+- A description should be provided for all identified features.
+- An asset describing the ways to test the presence of each control should be documented in feature file with the name `<service-name>.feature`.
+- All higher-level groupings in the hierarchy should be completed with `taxonmy.md` and `<service-name>.feature` files of its own containing the common features.
```
```suggestion
- All higher-level groupings in the hierarchy have been completed with `taxonmy.md` and `<service-name>.feature` files of its own containing the common features.
```

eddie-knight replied:
It looks as if we have diverged on the naming convention between the .md and .feature files.

What do you think of using a unique name for each taxonomy file, similar to the gherkin?

Alternatively it might be the other way around, with generic names for the feature files within a service directory.

eddie-knight commented on docs/governance/community-policies/definition-of-done.md:
```diff
@@ -0,0 +1,14 @@
+- All identified features should be numbered according to the _common-cloud-control_ guidelines.
+- A description should be provided for all identified features.
+- An asset describing the ways to test the presence of each control should be documented in feature file with the name `<service-name>.feature`.
+- All higher-level groupings in the hierarchy should be completed with `taxonomy.md` and `<service-name>.feature` files containing the common features.
```
```suggestion
- All higher-level groupings in the hierarchy have been completed with `taxonomy.md` and `<service-name>.feature` files containing the common features.
```

eddie-knight replied:
It looks as if we have diverged on the naming convention between the .md and .feature files.

What do you think of using a unique name for each taxonomy file, similar to the gherkin?

Alternatively it might be the other way around, with generic names for the feature files within a service directory.

smendis-scottlogic replied:
<service-name-short>-taxonomy
will be considered as the file name for bothe md and gherkin files.

E.g.: rdms-taxonomy.md & rdms-taxonomy.feature

eddie-knight reviewed (changes requested):
Hey @smendis-scottlogic, thanks again for doing this. 

Could you review the feedback bombs I dropped on here?

eddie-knight commented on docs/governance/community-policies/numbering-format.md:
```diff
@@ -0,0 +1,13 @@
+
+This is the proposed numbering format used in the taxonomy service features.
+
+`CCC.TX.CAT.XX.YYY`
```
It occurs to me that we can have a unique service identifier such as `OBST` for Object Storage, and it mitigates the need for inclusion of a category number.

Also, if we use C/F to specify if the final value is a control or feature ID, then we mitigate the need for `TX`. 

**CCC.`Service** Name Shorthand`.F`Feature Incrementer`**

example: `CCC.OBST.F01`

What do you think of this idea?

sshiells-scottlogic replied:
I like it. Though it does raise a question for me. CCC is my first FINOS project so I am not too familiar with what else it does. But are there are any other FINOS projects that do something similar? 

Should a standard identifier format be something that is consistent across the FINOS level? Or do you think that would be too wide a scope to capture.

e.g.
<FINOS_PROJECT_NAME>.<PROJECT_SPECIFC_IDS>.<FEATURE_INCREMENTER>

FINOS_PROJECT_NAME and FEATURE_INCREMENTER are a standard approach taken by all FINOS projects, and individiual projects are free to specify their PROJECT_SPECIFIC_IDS?

eddie-knight replied:
I doubt it, but let's ask around.

@robmoffat @karlmoll Do you have visibility to answer this? Has FDC3, TraderX, or anything else created a unique identifier pattern that we could draw inspiration from?

robmoffat replied:
The closest thing I can think is where in FDC3 we named all of our different tests, since they exercised different features of FDC3.  e.g. https://github.com/finos/FDC3/blob/main/toolbox/fdc3-conformance/Open-Tests.md

You can see on here we have `AOpensBWithContext1` or `AFailsToOpenB3` etc.  Since you'll also be building tests in CCC/CFI you'll definitely need some identifiers to separate them out.  The one piece of advice I can give is that numbers aren't super-helpful and we did get into some renumbering issues at times, which you don't get with identifiers.

damienjburks commented on docs/governance/community-policies/numbering-format.md:
```diff
@@ -0,0 +1,13 @@
+
+`CCC.TX.CAT.XX.YYY`
+
+- **CCC** - Common Coloud Control. Fixed
```
```suggestion
- **CCC** - Common Cloud Control. Fixed
```

smendis-scottlogic commented:
> Hey @smendis-scottlogic, thanks again for doing this.
> 
> Could you review the feedback bombs I dropped on here?

@eddie-knight addresses the review comments, can you take a look when possible? Thanks!
